src/tests/components/CourseDetails.test.js

```diff
@@ -12,43 +13,123 @@ describe("Course details", () => {
 
         expect(tagList).toBeEmptyDOMElement();
     });
+    */
 
-    it("displays award tag", () => {
+    it("displays level and award tag when both level and award are present", () => {
         const exampleCourse = {
+            level: "undergraduate",
             award: "BA (Hons)",
         };
 
         render(<CourseDetails course={exampleCourse} />);
 
-        const tag = screen.getByRole("listitem");
+        const tag = screen.getAllByRole("listitem");
 
-        expect(tag).toHaveTextContent("BA (Hons)");
-        expect(within(tag).getByTestId("tag-icon")).toHaveClass("c-icon--university");
+        expect(tag[0]).toHaveTextContent("undergraduate");
+        expect(tag[0]).toHaveTextContent("BA (Hons)");
+        expect(within(tag[0]).getByTestId("tag-icon")).toHaveClass("c-icon--mortar-board");
     });
 
-    it("displays start date tag", () => {
+    it("displays level and award tag when both only level is present", () => {
+        const exampleCourse = {
+            level: "undergraduate",
+        };
+
+        render(<CourseDetails course={exampleCourse} />);
+
+        const tag = screen.getAllByRole("listitem");
+
+        expect(tag[0]).toHaveTextContent("undergraduate");
+        expect(tag[0]).not.toHaveTextContent("BA (Hons)");
+        expect(within(tag[0]).getByTestId("tag-icon")).toHaveClass("c-icon--mortar-board");
+    });
+
+    it("displays level and award tag when only award is present", () => {
+        const exampleCourse = {
+            award: "BA (Hons)",
+        };
+
+        render(<CourseDetails course={exampleCourse} />);
+
+        const tag = screen.getAllByRole("listitem");
+
+        expect(tag[0]).not.toHaveTextContent("undergraduate");
+        expect(tag[0]).toHaveTextContent("BA (Hons)");
+        expect(within(tag[0]).getByTestId("tag-icon")).toHaveClass("c-icon--mortar-board");
+    });
+
+    it("does not display level and award tag when neither level or award is present", () => {
+        const exampleCourse = {};
+
+        render(<CourseDetails course={exampleCourse} />);
+
+        const tag = screen.queryAllByRole("listitem");
+
+        if (tag.length === 0) {
+            expect((tag.length = 0));
+        } else {
```

Comment:
I don't understand this bit. Presumably `expect((tag.length = 0));` is doing nothing because it's assigning `0` to `tag.length`, and that's not something I'd expect to verify with `expect`. Can this be simplified to:
```suggestion
        if (tag.length > 0) {
```
(I don't know whether it can BTW)

Aside from the above - I think conditional checks in tests are a bit strange and I think we should avoid them. As you say, this is going to be refactored at some point. Although I wouldn't want it to be merged in to `dev` like this if we can avoid it.